pkgs/development/compilers/cudatoolkit/auto-add-opengl-runpath-hook.sh

```diff
@@ -1,14 +1,22 @@
-# Run autoOpenGLRunpath on all files
+# Run addOpenGLRunpath on all dynamically linked, ELF files
 echo "Sourcing auto-add-opengl-runpath-hook"
 
-autoAddOpenGLRunpathPhase  () {
-    # TODO: support multiple outputs
-    for file in $(find ${out,lib,bin} -type f); do
-        addOpenGLRunpath $file
-    done
-}
+autoAddOpenGLRunpathPhase() (
+  local outputPaths=($(for o in $(getAllOutputNames); do echo "${!o}"; done))
+  find "${outputPaths[@]}" -type f -executable -print0  | while IFS= read -rd "" f; do
+    if isELF "$f"; then
+      if patchelf --print-needed "$f" >/dev/null 2>&1; then
```

Comment:
Is it the exit code that's significant here? Can we add a comment about how this works/what is the purpose of the test/what does the exit code mean?

Reply:
I switched to `--print-interpreter` and added a comment.

```
❯ cat main.go
package main
func main() {}

❯ go build main.go
❯ patchelf --print-interpreter main; echo $?
patchelf: cannot find section '.interp'. The input file is most likely statically linked
1

❯ patchelf --print-interpreter $(realpath $(which patchelf)); echo $?
/nix/store/vfmkdh2s2cnnk3igcn0bjpx38kl3zw8a-glibc-2.37-8/lib/ld-linux-x86-64.so.2
0
```